packages/ui-components/src/components/flow-canvas/canvas-controls/canvas-controls.tsx

```diff
@@ -0,0 +1,95 @@
+import { useReactFlow } from '@xyflow/react';
+import { t } from 'i18next';
+import { RefreshCcw, ZoomIn, ZoomOut } from 'lucide-react';
+import { useCallback } from 'react';
+import { Button } from '../../../ui/button';
+import { Tooltip, TooltipContent, TooltipTrigger } from '../../../ui/tooltip';
+
+import { CenterFlowIcon } from '../../../icons';
+import { VerticalDivider } from '../../../ui/vertical-divider';
+import { PanningModeControls } from './panning-mode-controls';
+
+const CanvasControls = ({ topOffset }: { topOffset?: number }) => {
+  const reactFlow = useReactFlow();
+
+  const handleZoomIn = useCallback(() => {
+    reactFlow.zoomIn({
+      duration: 200,
+    });
+  }, [reactFlow]);
+
+  const handleZoomOut = useCallback(() => {
+    reactFlow.zoomOut({
+      duration: 200,
+    });
+  }, [reactFlow]);
+
+  const handleZoomReset = useCallback(() => {
+    reactFlow.zoomTo(1);
+  }, [reactFlow]);
+
+  const handleFitToView = useCallback(async () => {
+    await reactFlow.fitView({
+      nodes: reactFlow.getNodes().slice(0, 5),
+      minZoom: 0.5,
+      maxZoom: 1.2,
```

Comment:
if you are adding a new consts file, would be cool to have also those values in there